cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptFilterTransform.java

```diff
@@ -98,6 +102,10 @@ private void init() {
     ScriptEngineManager manager = new ScriptEngineManager();
     engine = manager.getEngineByName("JavaScript");
 
+    // initialize metrics and logging
+    engine.put("metrics", metrics);
+    engine.put("LOG", logger);
```

Comment:
In general, there are two ways to do this. One is through a global context object (like `CDAP` I mentioned above). The other is through a context object passed to the function. Personally I like the function argument better since using global is always less preferred.